### PR TITLE
Test transform of NationalRail data

### LINKED ISSUE #37: Transform NationalRail tests
### Task
- [x] Write tests for transforming national rail xml data

### User Story
As a developer I want to test the transformation code to ensure that it works as expected.

### PR DESCRIPTION
**Pull request by FarihaChoudhury**
Tests the transformation functions for NationalRail data. Created a test class and ```test_data.xml``` file containing 2 example incidents. 

closes #37

### CODE REVIEW

File: national_rail/transform.py [hunk 1]
```diff
@@ -133,25 +133,35 @@ def process_pt_incidents(incidents: list[ET.Element], namespaces: dict) -> list[
...
-    incidents_dataset = process_pt_incidents(all_incidents, nr_namespaces)
+    return incidents_data
 
-    logging.info("Transformation of NationalRail completed successfully")
 
-    return incidents_dataset
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO,
+                        format="%(asctime)s - %(levelname)s - %(message)s")
+    transform()
```
Comment (Zhi-704) [inline national_rail/transform.py]: Good call basic config should be called outside of functions

File: national_rail/transform.py [hunk 2]
```diff
@@ -133,25 +133,35 @@ def process_pt_incidents(incidents: list[ET.Element], namespaces: dict) -> list[
...
+        relevant data, populates a list of dictionary and returns it."""
+
+    national_rail_tree_root = reverse_tree(load_tree_root(national_rail_xml))
+    all_incidents = get_incidents(national_rail_tree_root, namespace)
+    incidents_dataset = process_pt_incidents(all_incidents, namespace)
+
+    return incidents_dataset
+
+
 def transform():
```
Comment (Zhi-704) [inline national_rail/transform.py]: Type hints?

File: national_rail/transform.py [hunk 3]
```diff
@@ -133,25 +133,35 @@ def process_pt_incidents(incidents: list[ET.Element], namespaces: dict) -> list[
     return dataset
 
 
+def transform_xml_file(national_rail_xml: str, namespace: dict):
```
Comment (Zhi-704) [inline national_rail/transform.py]: Missing a type hint here as well

File: national_rail/transform.py (hunk 2 above)
Comment (FarihaChoudhury) [reply national_rail/transform.py]: yes thanks

File: national_rail/transform.py (hunk 3 above)
Comment (FarihaChoudhury) [reply national_rail/transform.py]: yes

**Review by Zhi-704 — APPROVED** [review]
Looks good! Approved

**Pull request merged**